python/ray/data/_internal/execution/bundle_queue/bundle_queue.py

```diff
@@ -22,22 +22,33 @@ def add(self, bundle: "RefBundle") -> None:
         ...
 
     @abc.abstractmethod
-    def pop(self) -> "RefBundle":
+    def get_next(self) -> "RefBundle":
         """Remove and return the head of the queue.
 
         Raises:
             IndexError: If the queue is empty.
+
+        Returns:
+            A Refbundle if has_next() is True
         """
         ...
 
     @abc.abstractmethod
-    def peek(self) -> Optional["RefBundle"]:
+    def peek_next(self) -> Optional["RefBundle"]:
         """Return the head of the queue without removing it.
 
         If the queue is empty, return `None`.
         """
         ...
 
+    @abc.abstractmethod
+    def has_next(self) -> bool:
+        """Check if the queue has a valid bundle.
+
+        A valid bundle is one that has all its objects in the object store.
+        """
+        ...
```

Comment:
+1 to https://github.com/ray-project/ray/pull/54710/files#r2214058901

```suggestion
    def has_next(self) -> bool:
        """Check if the queue has a valid bundle."""
        ...
```
Let's allow subclasses to define their notion of valid